src/ReplicatedStorage/Aero/Shared/Signal.lua

```diff
@@ -0,0 +1,151 @@
+-- Signal
+-- Stephen Leitnick
+-- Based off of Anaminus' Signal class: https://gist.github.com/Anaminus/afd813efc819bad8e560caea28942010
+
+--[[
+
+	signal = Signal.new()
+
+	signal:Fire(...)
+	signal:Wait()
+	signal:WaitPromise()
+	signal:Destroy()
+	signal:DisconnectAll()
+	
+	connection = signal:Connect(functionHandler)
+
+	connection.Connected
+	connection:Disconnect()
+	connection:IsConnected()
+
+--]]
+
+local Promise
+
+local Connection = {}
+Connection.__index = Connection
+
+function Connection.new(signal, connection)
+	local self = setmetatable({
+		_signal = signal;
+		_conn = connection;
+		Connected = true;
+	}, Connection)
+	return self
+end
+
+function Connection:Disconnect()
+	if (self._conn) then
+		self._conn:Disconnect()
+		self._conn = nil
+	end
+	if (not self._signal) then return end
+	self.Connected = false
+	local connections = self._signal._connections
+	for i,c in ipairs(connections) do
```

Comment:
Couldn't this be replaced with table.find? Should run faster and possibly be more readable.

Reply:
Yes, `table.find` is better. I'll update that